### PR TITLE
Release v2.17.0

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,46 @@
...
+  - **aviatrix_fqdn_tag_rule**        
+3. Implemented new resource to decouple ``policy`` out of ``aviatrix_firewall``:
+  - **aviatrix_firewall_policy**        
+4. Implemented new resources to support Multi-Cloud Segmentation:
+  - **aviatrix_segmentation_security_domain**        
+  - **aviatrix_segmentation_security_domain_connection_policy**        
+  - **aviatrix_segmentation_security_domain_association**        
+5. Implemented support of Editing for **aviatrix_saml_endpoint**
+6. Implemented support of ``subnet_size`` and ``num_of_subnet_pairs`` for advanced option for **aviatrix_vpc** and **aviatrix_vpc** data source
+7. Implemented support of ``enable_multicast`` as switch to enable/disable multi-case for **aviatrix_aws_tgw**
```
Comment (CyrusJavan) [inline CHANGELOG.md]: typo: multi-case -> multicast

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,46 @@
...
+8. Implemented support of insame mode for GCP provider for **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (CyrusJavan) [inline CHANGELOG.md]: typo: insame -> insane

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,46 @@
...
+9. Implemented support of updating ``customized_routes`` and ``customized_route_advertisement`` for **aviatrix_aws_tgw**
+10. Implemented support of ``zone`` and ``ha_zone`` for Azure provider for **aviatrix_transit_gateway**
+11. Implemented support of ``zone`` and ``peering_ha_zone`` for Azure provider for **aviatrix_gateway**
+12. Implemented support of ``zone`` for Azure provider for **aviatrix_firewall_instance**
+13. Implemented new resource to decouple attaching **aviatrix_spoke_gateway** to **aviatrix_transit_gateway** out of **aviatrix_spoke_gateway**
```
Comment (CyrusJavan) [inline CHANGELOG.md:30]: You say we implemented a new resource here but don't say the name of the resource

**Comment by CyrusJavan** [comment]
I do not see the new `resource_group` attribute mentioned that was added in #517

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,47 @@
...
+  - **aviatrix_device_virtual_wan_attachment**
+2. Implemented new resource to decouple ``domain_names`` out of ``aviatrix_fqdn``:
+  - **aviatrix_fqdn_tag_rule**        
+3. Implemented new resource to decouple ``policy`` out of ``aviatrix_firewall``:
+  - **aviatrix_firewall_policy**        
+4. Implemented new resources to support Multi-Cloud Segmentation:
+  - **aviatrix_segmentation_security_domain**        
+  - **aviatrix_segmentation_security_domain_connection_policy**        
+  - **aviatrix_segmentation_security_domain_association**        
+5. Implemented support of Editing for **aviatrix_saml_endpoint**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for updating **aviatrix_saml_endpoint**

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,47 @@
-## 2.16.3 (Unreleased)
+## 2.17.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-6.2** (tested on **UserConnect-6.2.1700**)
+- Supported Terraform version: **v0.12.x**
+
+### Features:
```
Comment (CyrusJavan) [inline CHANGELOG.md:6]: I think we need to add this as well
```
Implemented support of ``resource_group`` for Azure provider in data source **aviatrix_vpc**
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,47 @@
...
+2. Implemented new resource to decouple ``domain_names`` out of ``aviatrix_fqdn``:
+  - **aviatrix_fqdn_tag_rule**        
+3. Implemented new resource to decouple ``policy`` out of ``aviatrix_firewall``:
+  - **aviatrix_firewall_policy**        
+4. Implemented new resources to support Multi-Cloud Segmentation:
+  - **aviatrix_segmentation_security_domain**        
+  - **aviatrix_segmentation_security_domain_connection_policy**        
+  - **aviatrix_segmentation_security_domain_association**        
+5. Implemented support of Editing for **aviatrix_saml_endpoint**
+6. Implemented support of ``subnet_size`` and ``num_of_subnet_pairs`` for advanced option for **aviatrix_vpc** and **aviatrix_vpc** data source
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for advanced options to specify of ``subnet_size`` and ``num_of_subnet_pairs`` for AWS, AWSGov, and Azure VPCs/VNets in **aviatrix_vpc** resource and data source

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,47 @@
...
+7. Implemented support of ``enable_multicast`` as switch to enable/disable multi-cast for **aviatrix_aws_tgw**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for launching AWS TGWs with Multicast capability through the ``enable_multicast`` attribute for **aviatrix_aws_tgw** resource

(maybe you can link the release notes https://docs.aviatrix.com/HowTos/UCC_Release_Notes.html ?)

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,47 @@
...
+8. Implemented support of insane mode for GCP provider for **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented Insane Mode support for GCP **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,47 @@
...
+9. Implemented support of updating ``customized_routes`` and ``customized_route_advertisement`` for **aviatrix_aws_tgw**
+10. Implemented support of ``zone`` and ``ha_zone`` for Azure provider for **aviatrix_transit_gateway**
+11. Implemented support of ``zone`` and ``peering_ha_zone`` for Azure provider for **aviatrix_gateway**
+12. Implemented support of ``zone`` for Azure provider for **aviatrix_firewall_instance**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for Availability Zone selection for the following resources in Azure:
- ``zone`` and ``peering_ha_zone`` for **aviatrix_gateway**
- ``zone`` and ``ha_zone`` for **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
- ``zone`` for **aviatrix_firewall_instance**

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,47 @@
...
+13. Implemented new resource to decouple attaching **aviatrix_spoke_gateway** to **aviatrix_transit_gateway** out of **aviatrix_spoke_gateway**
+  - **aviatrix_spoke_transit_attachment**        
+14. Implemented support of ``enable_egress_transit_firenet`` for **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Implemented support for Dual Transit FireNet workflow through the ``enable_egress_transit_firenet`` attribute for **aviatrix_transit_gateway**

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,4 +1,47 @@
...
+
+### Enhancements:
+1. Blocked updating ``allocate_new_eip``, ``eip`` and ``ha_eip`` for **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+2. Added ``private_subnets`` and ``public_subnets`` for **aviatrix_vpc** and **aviatrix_vpc** data source
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Added ``private_subnets`` and ``public_subnets`` as outputs for **aviatrix_vpc** resource and data source

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,4 +1,47 @@
...
+
+### Bug Fixes:
+1. Fixed issue where there is a diff after creating a GCP **aviatrix_vpc**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Fixed issue where there was a delta in state after creating a GCP **aviatrix_vpc**

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,4 +1,47 @@
...
+2. Fixed import issue for **aviatrix_firewall**
+3. Fixed issue where the metadata as text is too long by updating from "get" to "post" method
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Fixed issue where long metadata text was unable to be handled in **aviatrix_saml_endpoint** by updating operations from GET to POST method

**Review by CyrusJavan — APPROVED** [review]

**Pull request merged**